pkg/plugins/plugins.go

```diff
@@ -0,0 +1,86 @@
+package plugins
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+)
+
+var (
+	ErrNotImplements = errors.New("Plugin not implements particular driver")
+)
+
+type plugins struct {
+	sync.Mutex
+	plugins map[string]*Plugin
+}
+
+var storage = plugins{plugins: make(map[string]*Plugin)}
+
```

Comment:
@tiborvass @calavera To be clear, regardless of lazy or active loading of the plugin (there are reasoning for both approach), I would like to see https://github.com/mavenugo/docker/commit/7fc16f40de9c86ef89658043f19e702aff7c25de included so that network remote drivers can be integrated with a minimal impact to the overall plugin approach.